data-products/src/freestar_revenue_reporting/freestar_reporting_flow.py

```diff
@@ -0,0 +1,277 @@
+    network STRING,
+    url STRING,
+    utm_campaign STRING,
+    utm_content STRING,
+    utm_medium STRING,
+    utm_source STRING,
+    utm_term STRING,
+    site_domain STRING,
+    ad_unit STRING,
+    size STRING,
+    country_code STRING,
+    device_type STRING,
+    device_os STRING,
+    browser STRING,
+    integration_partner STRING,
+    impressions INTEGER,
+    net_revenue INTEGER,
+    net_cpm FLOAT
+);
+"""
+
+format_file_sql = f"""
+CREATE OR REPLACE FILE FORMAT {snowflake_table}_format
+TYPE = parquet;
+"""
+
+create_stage_sql = f"""
+CREATE OR REPLACE STAGE {snowflake_table}_stage
+FILE_FORMAT = {snowflake_table}_format;
+"""
+
+put_parquet_sql = f"""
+PUT file://{subdirectory}/{output_parquet_filename} @{snowflake_table}_stage
+"""
+
+load_sql = f"""
+COPY INTO {snowflake_table}
+    FROM (SELECT $1:record_date::DATE,
+                 $1:network::STRING,
+                 $1:url::STRING,
+                 $1:utm_campaign::STRING,
+                 $1:utm_content::STRING,
+                 $1:utm_medium::STRING,
+                 $1:utm_source::STRING,
+                 $1:utm_term::STRING,
+                 $1:site_domain::STRING,
+                 $1:ad_unit::STRING,
+                 $1:size::STRING,
+                 $1:country_code::STRING,
+                 $1:device_type::STRING,
+                 $1:device_os::STRING,
+                 $1:browser::STRING,
+                 $1:integration_partner::STRING,
+                 $1:impressions::INTEGER,
+                 $1:net_revenue::INTEGER,
+                 $1:net_cpm::FLOAT
+        FROM @{snowflake_table}_stage/data_combined.parquet
+        );
+"""
+
+# Define the Prefect flow
+@flow(name="Freestar Report Flow")
+async def freestar_report_flow():
+    extract_freestar_data()
+    await snowflake_query_sync(
+        query=create_table_sql,
+        snowflake_connector=PktSnowflakeConnector()
+    )
+    await snowflake_query_sync(
+        query=format_file_sql,
+        snowflake_connector=PktSnowflakeConnector()
+    )
+    await snowflake_query_sync(
+        query=create_stage_sql,
+        snowflake_connector=PktSnowflakeConnector()
+    )
+    await snowflake_query_sync(
+        query=put_parquet_sql,
+        snowflake_connector=PktSnowflakeConnector()
+    )
+    await snowflake_query_sync(
+        query=load_sql,
+        snowflake_connector=PktSnowflakeConnector()
+    )
+
+FLOW_SPEC = FlowSpec(
+    flow=freestar_report_flow,
+    docker_env="base",
+    secrets=[
+        FlowEnvar(
+            envar_name="DF_CONFIG_SNOWFLAKE_CREDENTIALS",
+            envar_value=f"data-flows/{CS.deployment_type}/snowflake-credentials",
+        ),
+        FlowEnvar(
+            envar_name="FREESTAR_CREDENTIALS",
+            envar_value=f"data-flows/{CS.deployment_type}/freestar-credentials",
+        )
+    ],
+    deployments=[
+        FlowDeployment(
+            deployment_name="freestar_extraction",
+            schedule=CronSchedule(cron="0 10 * * *"),
```

Comment:
@mmiermans honestly I chose that time because we are pulling data from Freestar's "yesterday". I'm not clear on how they define "yesterday"...so was making it later in the morning. but that brings up a question I have...are prefect flows run on UTC? I assumed so...which would have this running early morning.

Reply:
I think this is definitely something we can talk about.  For now I have created https://mozilla-hub.atlassian.net/browse/DENG-1630 as an Epic to potentially implement an orchestration flow.